### ISSUE TITLE
`useDescendent` fails to build with TypeScript 4.8.3

### ISSUE THREAD

**Issue by vicary** [chakra-ui/chakra-ui#6671]
### Description

When I use TypeScript 4.8.3 I expect @chakra-ui to build.

### Link to Reproduction

N/A

### Steps to reproduce

1. Install `@chakra-ui/react`
2. Upgrade TypeScript to 4.8.3
3. See additional information below

### Chakra UI Version

master / latest

### Browser

N/A

### Operating System

- [X] macOS
- [ ] Windows
- [X] Linux

### Additional Information

I am the maintainer of `gqty` and we are using `@chakra-ui` in our examples and website.

When we try to upgrade to TypeScript 4.8.3 in https://github.com/gqty-dev/gqty/pull/1096, our next.js deployment starts producing a type error from the internal hook `useDescdendants`.

Apologies for I am not familiar enough with @chakra-ui to know which component is using this hook internally, let me share the build error below.

```
Type error: Type 'K' does not satisfy the constraint 'Record<string, any>'.
```

<img width="1079" alt="image" src="https://user-images.githubusercontent.com/85772/189515635-e66457c0-60fd-4f51-8a72-ef1ce34c0408.png">

We are not ready to move to @chakra-ui / React 18 just yet, so we would like to stay at 1.x for the moment. But this issue exists in both 1.x, 2.x and the current master.

**Comment by AlexandruCalinica** [comment]
I am having this issue with 
`@chakra-ui/react`: "^1.8.8"
`typescript: "^4.6.4"`

**Comment by banuni** [comment]
> I am having this issue with `@chakra-ui/react: "^1.8.8"` `typescript: "^4.6.4"`

to workaround this you'll have to freeze the 4.7 version, something like `"typescript": "~4.7.0"` should work

**Comment by segunadebayo** [comment]
I just released [v1.8.9](https://www.npmjs.com/package/@chakra-ui/react/v/1.8.9)

Please update this version recent version to resolve the problem.

**Comment by josealonso** [comment]
Same issue here using *chakra 1.8.8* and the *AlertDialog* component.
